lib/bsdlib/nrf91_sockets.c

```diff
@@ -827,6 +844,17 @@ static int nrf91_socket_offload_getaddrinfo(const char *node,
 	}
 	int retval = nrf_getaddrinfo(node, service, nrf_hints_ptr, &nrf_res);
 
+	if (retval != 0) {
+		error = nrf_to_z_error_code(retval);
+		if (error == DNS_EAI_SYSTEM) {
+			errno = retval;
```

Comment:
We should not set retval (which is in `NRF_*` error space into the system-wide `errno` variable). Perhaps it would be possible to re-use `bsd_os_errno_set` function to set the correct, translated errno value?

Reply:
Fixed this